backend/app/services/implementations/log_records_service.py

```diff
@@ -24,20 +26,31 @@ def __init__(self, logger):
 
     def add_record(self, log_record):
         new_log_record = log_record.copy()
-
+        residents = new_log_record["residents"]
         tag_names = new_log_record["tags"]
+
+        del new_log_record["residents"]
         del new_log_record["tags"]
 
         try:
             new_log_record = LogRecords(**new_log_record)
+            self.construct_residents(new_log_record, residents)
             self.construct_tags(new_log_record, tag_names)
 
             db.session.add(new_log_record)
             db.session.commit()
-            return log_record
+            return {**log_record, "residents": residents}
         except Exception as postgres_error:
             raise postgres_error
         
+    def construct_residents(self, log_record, residents):
+        for resident_id in residents:
```

Comment:
This isn't currently done in tags (TBD), but I realized that if we pass in the same resident ID multiple times, it creates multiple records for the same resident. To solve this you can use a set to help:
` resident_ids = list(set(resident_ids))`